packages/axios-extension/src/auth/index.ts

```diff
@@ -49,3 +51,55 @@ export function attachUaaAuthInterceptor(
         return request;
     });
 }
+
+/**
+ * Get the reentrace ticket from the backend and add it to the header.
+ *
+ * @param options
+ * @param options.provider an instance of an ABAP service provider
+ */
+export function attachReentranceTicketAuthInterceptor({ provider }: { provider: ServiceProvider }): void {
+    /* This cannot be a const like eslint claims */
+    /* eslint-disable prefer-const */
+    let oneTimeInterceptorId: number;
+    const ejectCallback = () => provider.interceptors.request.eject(oneTimeInterceptorId);
+    oneTimeInterceptorId = provider.interceptors.request.use(
+        getReentranceTicketAuthInterceptor({ provider, ejectCallback })
+    );
+}
+
+/**
+ * Get the interceptor that fetches and uses reentrance tickets from the backend.
+ *
+ * @param options
+ * @param options.provider an instance of an ABAP service provider
+ * @param options.ejectCallback
+ * @returns the interceptor function to fetch and use reentrace tickets
+ */
+export function getReentranceTicketAuthInterceptor({
+    provider,
+    ejectCallback
+}: {
+    provider: ServiceProvider;
+    ejectCallback: () => void;
+}): (request: AxiosRequestConfig) => Promise<AxiosRequestConfig<any>> {
+    return async (request: AxiosRequestConfig) => {
+        const { reentranceTicket, apiUrl } = await getReentranceTicket({
+            backendUrl: provider.defaults.baseURL,
+            logger: provider.log
+        });
+        if (apiUrl && apiUrl != provider.defaults.baseURL) {
+            // Reentrance tickets work with API hostnames. If the original URL was not one, this will replace it
+            // with the API hostname returned
+            provider.log.warn(
+                `Replacing provider's default base URL (${provider.defaults.baseURL}) with API URL: ${apiUrl}`
+            );
+            provider.defaults.baseURL = apiUrl;
```

Comment:
This might be a problem if the first call is not executed on the `provider` but on a service created based on the provider because when we create an instance of a service (in `ServiceProvider.createService`) we copy the interceptors over to the service, however, the function here would modify the `baseUrl` of the provider but not affecting the service `baseUrl`.
Thinking about it, this is not a special problem here but more of a conceptual problem in the provider/service concept.